lib/timers.js

```diff
@@ -317,51 +322,76 @@ exports.enroll = function(item, msecs) {
  */
 
 
-exports.setTimeout = function(callback, after) {
+exports.setTimeout = function(callback, after, arg1, arg2, arg3) {
   if (typeof callback !== 'function') {
     throw new TypeError('"callback" argument must be a function');
   }
 
-  after *= 1; // coalesce to number or NaN
-
-  if (!(after >= 1 && after <= TIMEOUT_MAX)) {
-    after = 1; // schedule on next tick, follows browser behaviour
+  var len = arguments.length;
```

Comment:
Can we check for typeof args1 == 'undefined' rather than using length? Or is undefined a valid value for args1?